v2/internal/config/vars_test.go

```diff
@@ -80,3 +80,52 @@ func Test_Cloud_ResourceManagerAudienceSet(t *testing.T) {
 	g.Expect(cld.Services[cloud.ResourceManager].Endpoint).To(Equal(cloud.AzurePublic.Services[cloud.ResourceManager].Endpoint))
 	g.Expect(cld.Services[cloud.ResourceManager].Audience).To(Equal(cfg.ResourceManagerAudience))
 }
+
+func TestValidate(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	vOk := config.Values{
+		PodNamespace:            "test-namespace",
+		MaxConcurrentReconciles: 1,
+		OperatorMode:            3,
+		TargetNamespaces:        []string{},
+		DefaultReconcilePolicy:  "detach-on-delete",
+	}
+
+	g.Expect(vOk.Validate()).NotTo(HaveOccurred())
+
+	vKoNamespace := config.Values{
+		PodNamespace: "",
+	}
+
+	g.Expect(vKoNamespace.Validate().Error()).Error().Should(Equal("missing value for POD_NAMESPACE"))
```

Comment:
This assertion will blow up with a panic if `vKoNamespace.Validate()` fails to return an error. Plus, asserting on errors this way is brittle as any wording change to the error will break the test entirely. 

Suggested refactor:
```suggestion
	err := vKoNamespace.Validate()
	g.Expect(err).To(MatchError(ContainSubstring("POD_NAMESPACE")))
	g.Expect(err).To(MatchError(ContainSubstring("missing value"))
```
The `MatchError()` predicate only passes if an error is both present _and_ the error string matches the provided subpredicate.

Reply:
@theunrepentantgeek, at this stage I wonder if we should just check if an error is thrown when namespace is empty or if you feel comfortable in keeping the error message consistent with the substrings.
In the meanwhile, I've amended the PR with the latest suggestions.